### DISCUSSION TITLE
Requesting assistance initiating alsa-scarlett-gui with 4g Solo

### DISCUSSION THREAD

**Discussion by a-star-full-of-skies** [geoffreybennett/alsa-scarlett-gui#111] [Q&A] [answered]
Firstly I'm grateful for your efforts Geoffrey, thank you kindly.  

I have a 4th gen Solo, and I felt very confident in purchasing it because the GUI interface supported it! 

Initially, when I load 0.4.0 (in OpenSUSE Tumbleweed, KDE, kernel 6.8.4, Wayland) I receive a message to download firmware.  I've downloaded the tarball, and put the bin files in /usr/lib/firmware/scarlett2, and i still receive the message 'Firmware Update Requred'.  I have tried to get the application to recognize the bin files, but I'm unable to get past the Firmware Update Required screen... 

Please help... I'm stuck

**Comment by geoffreybennett** [comment]
Hi @a-star-full-of-skies,

You're welcome. We should be able to get you going soon :crossed_fingers: :slightly_smiling_face:

Please run these commands and paste the output here to help us narrow down the problem:
- `alsa-scarlett-gui` (from the command-line, so you can see any output)
- `dmesg | grep -A 5 -B 5 -i focusrite` (remove the serial number from the output if you prefer)
- `ls -la /usr/lib/firmware/scarlett2`

Also download https://github.com/geoffreybennett/scarlett2, build it, run `./scarlett2 list-all` and paste the output here.

Thanks,
Geoffrey.

**Reply by a-star-full-of-skies** [reply]
Thanks a ton for your prompt response... I have high hopes too :crossed_fingers: :smile: 

ok so for ease of use, i went with the flatpak for the gui... so the command to run is different than a binary of course... `flatpak run vu.b4.alsa-scarlett-gui` ...and the output is simply "Firmware directory /usr/lib/firmware/scarlett2 does not exist"

This raises the question why is it not recognized since i can certainly go to /usr/lib/firmware/scarlett2 in my system, and running `ls` shows the scarlett2-*.bin collection.

Could this be a flatpak permissions issues? 

The output of dmesg | grep focusrite is as follows


```
[    2.305923] usb 1-7: Manufacturer: Keychron
[    2.429456] usb 1-8: new high-speed USB device number 4 using xhci_hcd
[    2.573394] usb 1-8: New USB device found, idVendor=1235, idProduct=8218, bcdDevice= 7.b6
[    2.573396] usb 1-8: New USB device strings: Mfr=1, Product=3, SerialNumber=2
[    2.573397] usb 1-8: Product: Scarlett Solo 4th Gen
[    2.573398] usb 1-8: Manufacturer: Focusrite
[    2.573398] usb 1-8: SerialNumber: xxxxxxxxxxxx
[    2.696148] usb 1-9: new full-speed USB device number 5 using xhci_hcd
[    2.841215] usb 1-9: New USB device found, idVendor=f1f1, idProduct=0315, bcdDevice= 0.01
[    2.841218] usb 1-9: New USB device strings: Mfr=1, Product=2, SerialNumber=3
[    2.841219] usb 1-9: Product: YD3xn15mx
--
[    8.191300] Bluetooth: hci0: Found Intel DDC parameters: intel/ibt-0041-0041.ddc
[    8.192879] Bluetooth: hci0: Applying Intel DDC parameters completed
[    8.195918] Bluetooth: hci0: Firmware timestamp 2024.8 buildtype 1 build 79483
[    8.195924] Bluetooth: hci0: Firmware SHA1: 0x7c7b67b9
[    8.198468] iwlwifi 0000:06:00.0: WRT: Invalid buffer destination
[    8.262287] usb 1-8: Focusrite Scarlett Gen 4 Mixer Driver enabled (pid=0x8218); report any issues to https://github.com/geoffreybennett/scarlett-gen2/issues
[    8.262531] usb 1-8: Firmware version 1974
[    8.263504] usb 1-8: Focusrite Scarlett Gen 4 firmware version 1974 is too old; need 2115
[    8.263934] usbcore: registered new interface driver snd-usb-audio
[    8.295649] Bluetooth: MGMT ver 1.22
[    8.311025] NET: Registered PF_ALG protocol family
[    8.355756] iwlwifi 0000:06:00.0: WFPM_UMAC_PD_NOTIFICATION: 0x20
[    8.355808] iwlwifi 0000:06:00.0: WFPM_LMAC2_PD_NOTIFICATION: 0x1
```

the output of ls /usr/lib/firmware/scarlett2 is

```
total 5680
drwxr-xr-x 1 root root    1198 Apr  8 20:43 .
drwxr-xr-x 1 root root   15350 Apr  8 17:31 ..
drwxr-xr-x 1 root root    1182 Apr  8 20:45 firmware
-rw-r--r-- 1 root root  125236 Feb 20 19:00 scarlett2-1235-8201-1653.bin
-rw-r--r-- 1 root root  105780 Feb 20 19:00 scarlett2-1235-8203-1583.bin
-rw-r--r-- 1 root root  104244 Feb 20 19:00 scarlett2-1235-8204-1583.bin
-rw-r--r-- 1 root root  122420 Feb 20 19:00 scarlett2-1235-8206-1246.bin
-rw-r--r-- 1 root root  120116 Feb 20 19:00 scarlett2-1235-8207-1195.bin
-rw-r--r-- 1 root root  123188 Feb 20 19:00 scarlett2-1235-8208-1195.bin
-rw-r--r-- 1 root root  123444 Feb 20 19:00 scarlett2-1235-820a-1993.bin
-rw-r--r-- 1 root root  120884 Feb 20 19:00 scarlett2-1235-820b-1955.bin
-rw-r--r-- 1 root root  123956 Feb 20 19:00 scarlett2-1235-820c-1955.bin
-rw-r--r-- 1 root root  107316 Feb 20 19:00 scarlett2-1235-8210-1605.bin
-rw-r--r-- 1 root root  107060 Feb 20 19:00 scarlett2-1235-8211-1605.bin
-rw-r--r-- 1 root root  156468 Feb 20 19:00 scarlett2-1235-8212-1605.bin
-rw-r--r-- 1 root root  169780 Feb 20 19:00 scarlett2-1235-8213-1605.bin
-rw-r--r-- 1 root root  178740 Feb 20 19:00 scarlett2-1235-8214-1605.bin
-rw-r--r-- 1 root root  225588 Feb 20 19:00 scarlett2-1235-8215-1644.bin
-rw-r--r-- 1 root root  166452 Feb 20 19:00 scarlett2-1235-8216-1769.bin
-rw-r--r-- 1 root root  215348 Feb 20 19:00 scarlett2-1235-8217-1769.bin
-rw-r--r-- 1 root root  171572 Feb 20 19:00 scarlett2-1235-8218-2115.bin
-rw-r--r-- 1 root root  199220 Feb 20 19:00 scarlett2-1235-8219-2128.bin
-rw-r--r-- 1 root root  206132 Feb 20 19:00 scarlett2-1235-821a-2108.bin
-rw-r--r-- 1 root root 2797323 Apr  8 18:13 scarlett2-firmware-2128b.tar.gz
```

output of scarlett2 list-all is

```
USB Product ID, Product Name, and Firmware versions available (* = connected)
 8203 Scarlett 2nd Gen 6i6     1583
 8204 Scarlett 2nd Gen 18i8    1583
 8201 Scarlett 2nd Gen 18i20   1653
 8211 Scarlett 3rd Gen Solo    1605
 8210 Scarlett 3rd Gen 2i2     1605
 8212 Scarlett 3rd Gen 4i4     1605
 8213 Scarlett 3rd Gen 8i6     1605
 8214 Scarlett 3rd Gen 18i8    1605
 8215 Scarlett 3rd Gen 18i20   1644
 8216 Vocaster One             1769
 8217 Vocaster Two             1769
*8218 Scarlett 4th Gen Solo    2115 (running: 1974)
 8219 Scarlett 4th Gen 2i2     2128
 821a Scarlett 4th Gen 4i4     2108
 8206 Clarett USB 2Pre         1246
 8207 Clarett USB 4Pre         1195
 8208 Clarett USB 8Pre         1195
 820a Clarett+ 2Pre            1993
 820b Clarett+ 4Pre            1955
 820c Clarett+ 8Pre            1955
```

**Comment by a-star-full-of-skies** [comment, marked as answer]
Geoffrey, i am now up and running with the command `scarlett2 -u` which updated the firmware (ooh extra air mode) and made the device recognizable to your gui

Thank you for your help... really great to have your work available, and some financial support is on the way to say thanks!

**Comment by geoffreybennett** [comment]
Awesome, and thank you very much for your support! It's truly appreciated.

I have created issue #112 to investigate how to fix the Flatpak version so that firmware updates can be done through it.

**Reply by a-star-full-of-skies** [reply]
Issue solved in 16 hours hehe you're a boss... I wouldn't be able to mess with Focusrite products if it weren't for your work in the first place... my delight to be able to support at that level.  Your work is very professional... impressed. Thanks again!

**Comment by geoffreybennett** [comment]
I've got a fix for this. Next version of the flatpak I release will include the firmware files so no need for a separate download.
